### PR TITLE
ClusterLoader - Adding test scenario to config

### PR DESCRIPTION
**Pull request by krzysied**
ref #641

### CODE REVIEW

**Comment by krzysied** [comment]
/hold
I want to check if summary names remain unchanged.

File: clusterloader2/pkg/config/cluster.go [hunk 1]
```diff
@@ -16,15 +16,18 @@ limitations under the License.
...
-	EnableExecService        bool          `json: enableExecService`
-	TearDownPrometheusServer bool          `json: tearDownPrometheusServer`
-	TestConfigPath           string        `json: testConfigPath`
-	TestOverridesPath        []string      `json: testOverrides`
+	ClusterConfig            ClusterConfig    `json: clusterConfig`
+	ReportDir                string           `json: reportDir`
+	EnablePrometheusServer   bool             `json: enablePrometheusServer`
+	EnableExecService        bool             `json: enableExecService`
+	TearDownPrometheusServer bool             `json: tearDownPrometheusServer`
+	TestScenario             api.TestScenario `json: testScenario`
```
Comment (wojtek-t) [inline clusterloader2/pkg/config/cluster.go]: Can we remove TestOverridesPath now?

File: clusterloader2/pkg/test/simple_test_executor.go [hunk 2]
```diff
@@ -86,8 +86,12 @@ func (ste *simpleTestExecutor) ExecuteTest(ctx Context, conf *api.Config) *error
 		if ctx.GetClusterLoaderConfig().ReportDir == "" {
 			klog.Infof("%v: %v", summary.SummaryName(), summary.SummaryContent())
 		} else {
+			id := ""
+			if ctx.GetClusterLoaderConfig().TestScenario.Identifier != "" {
+				id = "_" + ctx.GetClusterLoaderConfig().TestScenario.Identifier
```
Comment (wojtek-t) [inline clusterloader2/pkg/test/simple_test_executor.go]: Why the id starts with "_" ?

File: clusterloader2/pkg/config/cluster.go (hunk 1 above)
Comment (krzysied) [reply clusterloader2/pkg/config/cluster.go]: Yes, it's being removed (or rather moved to test scenario structure) by this PR.

File: clusterloader2/pkg/test/simple_test_executor.go (hunk 2 above)
Comment (krzysied) [reply clusterloader2/pkg/test/simple_test_executor.go]: I'm using it later in file name. I want to add additional separator "_" if id is not empty.
When identifier provided: SummaryName_configName_testId_timstamp.ext.
Without id: SummaryName_configName_timstamp.ext

Comment (wojtek-t) [reply clusterloader2/pkg/test/simple_test_executor.go]: well, id is not the most intuitive name then :)

infix? distinctor? (neither of those is good, but less confusing)

Comment (krzysied) [reply clusterloader2/pkg/test/simple_test_executor.go]: Changed to "testDistinctor".

File: clusterloader2/pkg/config/cluster.go (hunk 1 above)
Comment (wojtek-t) [inline clusterloader2/pkg/config/cluster.go]: Wait - TestScenario is not a flag to ClusterLoader (it's the path to config file with test scenario)

Comment (krzysied) [reply clusterloader2/pkg/config/cluster.go]: "testSuiteConfigPath" - flag - Path to the test suite config file.
"TestScenario" - structure - defines test config, overrides and identifier.

File: clusterloader2/pkg/config/cluster.go [hunk 3]
```diff
@@ -16,14 +16,19 @@ limitations under the License.
...
 type ClusterLoaderConfig struct {
-	ClusterConfig     ClusterConfig
-	ReportDir         string
-	EnableExecService bool
-	TestConfigPath    string
-	TestOverridesPath []string
-	PrometheusConfig  PrometheusConfig
+	ClusterConfig            ClusterConfig
+	ReportDir                string
+	EnablePrometheusServer   bool
```
Comment (wojtek-t) [inline clusterloader2/pkg/config/cluster.go]: seems unused

File: clusterloader2/pkg/config/cluster.go [hunk 4]
```diff
@@ -16,14 +16,19 @@ limitations under the License.
...
+	EnableExecService        bool
+	TearDownPrometheusServer bool
```
Comment (wojtek-t) [inline clusterloader2/pkg/config/cluster.go]: this one too

File: clusterloader2/pkg/test/simple_test_executor.go [hunk 5]
```diff
@@ -86,8 +86,19 @@ func (ste *simpleTestExecutor) ExecuteTest(ctx Context, conf *api.Config) *error
...
 			klog.Infof("%v: %v", summary.SummaryName(), summary.SummaryContent())
 		} else {
+			testDinstinctor := ""
+			if ctx.GetClusterLoaderConfig().TestScenario.Identifier != "" {
+				testDinstinctor = "_" + ctx.GetClusterLoaderConfig().TestScenario.Identifier
+			}
 			// TODO(krzysied): Remember to keep original filename style for backward compatibility.
-			filePath := path.Join(ctx.GetClusterLoaderConfig().ReportDir, summary.SummaryName()+"_"+conf.Name+"_"+summary.SummaryTime().Format(time.RFC3339)+"."+summary.SummaryExt())
+			filePath := path.Join(ctx.GetClusterLoaderConfig().ReportDir, summary.SummaryName()+
+				"_"+
```
Comment (wojtek-t) [inline clusterloader2/pkg/test/simple_test_executor.go]:
```
fileName := strings.Join([]string{summary.SummaryName(), conf.Name+testDistinctor, summary.SummaryTime.Format(...), "_")
filePath := path.Join(reportDir, strings.Join([]string{fileName, extension}, ".")
```

will be a bit more readable what is happening here

Comment (krzysied) [reply clusterloader2/pkg/test/simple_test_executor.go]: Done.

**Pull request merged**